Makefile

```diff
@@ -15,21 +23,32 @@ else
   WLARGS := -Wl,-rpath,"$(JULIA_DIR)/lib:$$ORIGIN"
 endif
 
-CFLAGS+=-O2 -fPIE -I$(JULIA_DIR)/include/julia
-LDFLAGS+=-L$(JULIA_DIR)/lib -L. -ljulia -lm $(WLARGS)
+CFLAGS+=-O2 -fPIE -I$(JULIA_DIR)/include/julia -I$(INCLUDE_DIR)
+LDFLAGS+=-L$(JULIA_DIR)/lib -L$(LIBDIR) -ljulia -lm $(WLARGS)
 
-.DEFAULT_GOAL := main
+.DEFAULT_GOAL := $(MAIN)
 
-libcg.$(DLEXT): build/build.jl src/CG.jl build/generate_precompile.jl build/additional_precompile.jl
+$(LIB_LIBCG) $(LIBCG_INCLUDES): build/build.jl src/CG.jl build/generate_precompile.jl build/additional_precompile.jl
 	$(JULIA) --startup-file=no --project=. -e 'using Pkg; Pkg.instantiate()'
 	$(JULIA) --startup-file=no --project=build -e 'using Pkg; Pkg.instantiate()'
-	$(JULIA) --startup-file=no --project=build $<
+	OUTDIR=$(OUTDIR) $(JULIA) --startup-file=no --project=build $<
 
-main.o: main.c
-	$(CC) $^ -c -o $@ $(CFLAGS) -DJULIAC_PROGRAM_LIBNAME=\"libcg.$(DLEXT)\"
+main.o: main.c $(LIBCG_INCLUDES)
+	$(CC) $< -c -o $@ $(CFLAGS) -DJULIAC_PROGRAM_LIBNAME=\"$(LIBCG)\"
 
-$(MAIN): main.o libcg.$(DLEXT)
+$(MAIN): main.o $(LIB_LIBCG)
 	$(CC) -o $@ $< $(LDFLAGS) -lcg
+ifeq ($(OS), Darwin)
+	# Make sure we can find and use the shared library on OSX
+	install_name_tool -change $(LIBCG) $(LIB_LIBCG) $@
+	## Alternatively, if we install the lib directory somewhere in the library
+	## search path, we could run
+	# install_name_tool -change $(LIBCG) @rpath/$(LIBCG) $@
+else ifeq ($(OS), WINNT)
+	echo "Please add $(LIBDIR) to your PATH before running $(MAIN)"
+else
+	echo "Please add $(LIBDIR) to LD_LIBRARY_PATH before running $(MAIN)"
```

Comment:
Can you link with `-Wl,-rpath,$(LIBDIR)` instead?

Reply:
I agree, rpath seems like a cleaner way to do this.

Reply:
Thanks, wasn't really sure how to do this.  Will fix.

Reply:
(Although now that I look at it, it makes perfect sense...)

Reply:
This is fixed for Linux (and OSX).  There doesn't seem to be an equivalent for Windows (according to the PackageCompiler documentation).

Reply:
I think Windows will look for dlls in the same directory as the executable, so setting `LIBDIR := $(OUTDIR)` might be an easy fix

Reply:
Okay, that makes sense.

Reply:
I did add `$(LIBDIR)` to the RPATH, and put all of the library outputs next to the executable (in `target/bin`) on Windows.